instances/treasury-devdao.near/widget/pages/settings/VotingDurationPage.jsx

```diff
@@ -0,0 +1,321 @@
+const instance = props.instance;
+if (!instance) {
+  return <></>;
+}
+
+const { treasuryDaoID } = VM.require(`${instance}/widget/config.data`);
+
+const daoPolicy = Near.view(treasuryDaoID, "get_policy", {});
+
+if (!daoPolicy) {
+  return <></>;
+}
+
+const lastProposalId = Near.view(treasuryDaoID, "get_last_proposal_id");
+
+const deposit = daoPolicy?.proposal_bond || 100000000000000000000000;
+
+const currentDurationDays =
+  Number(
+    daoPolicy.proposal_period.substr(
+      0,
+      daoPolicy.proposal_period.length - "000000000".length
+    )
+  ) /
+  (60 * 60 * 24);
+
+const [durationDays, setDurationDays] = useState(currentDurationDays);
+const [proposalsThatWillExpire, setProposalsThatWillExpire] = useState([]);
+const [showToastStatus, setToastStatus] = useState(null);
+const [isSubmittingChangeRequest, setSubmittingChangeRequest] = useState(false);
+
+const Container = styled.div`
+  font-size: 14px;
+  .border-right {
+    border-right: 1px solid rgba(226, 230, 236, 1);
+  }
+
+  .card-title {
+    font-size: 18px;
+    font-weight: 600;
+    padding-block: 5px;
+    border-bottom: 1px solid rgba(226, 230, 236, 1);
+  }
+
+  .selected-role {
+    background-color: rgba(244, 244, 244, 1);
+  }
+
+  .cursor-pointer {
+    cursor: pointer;
+  }
+
+  .tag {
+    background-color: rgba(244, 244, 244, 1);
+    font-size: 12px;
+    padding-block: 5px;
+  }
+
+  label {
+    color: rgba(153, 153, 153, 1);
+    font-size: 12px;
+  }
+
+  .fw-bold {
+    font-weight: 500 !important;
+  }
+
+  .p-0 {
+    padding: 0 !important;
+  }
+
+  .text-md {
+    font-size: 13px;
+  }
+
+  .theme-btn {
+    background-color: var(--theme-color) !important;
+    color: white;
+  }
+
+  .warning {
+    background-color: rgba(255, 158, 0, 0.1);
+    color: rgba(177, 113, 8, 1);
+    font-weight: 500;
+  }
+
+  .text-sm {
+    font-size: 12px !important;
+  }
+
+  .text-muted {
+    color: rgba(153, 153, 153, 1);
+  }
+
+  .text-red {
+    color: #d95c4a;
+  }
+
+  .toast {
+    background: white !important;
+  }
+
+  .toast-header {
+    background-color: #2c3e50 !important;
+    color: white !important;
+  }
+`;
+
+const ToastContainer = styled.div`
+  a {
+    color: black !important;
+    text-decoration: underline !important;
+    &:hover {
+      color: black !important;
+    }
+  }
+`;
+
+const cancelChangeRequest = () => {
+  setDurationDays(currentDurationDays);
+};
+
+const submitChangeRequest = () => {
+  setSubmittingChangeRequest(true);
+  Near.call({
+    contractName: treasuryDaoID,
+    methodName: "add_proposal",
+    deposit,
+    args: {
+      proposal: {
+        description: "Change proposal period",
+        kind: {
+          ChangePolicyUpdateParameters: {
+            parameters: {
+              proposal_period:
+                (60 * 60 * 24 * durationDays).toString() + "000000000",
+            },
```

Comment:
Don't we need to send all [parameters](https://github.com/near-daos/sputnik-dao-contract/blob/main/sputnikdao2/src/proposals.rs#L115)?
And I think we will need to update permissions as well (for all instances), because if I remember correctly `ChangePolicyUpdateParameters` isn't mentioned in policy which is set currently.

Reply:
When I look at the implementation of `update_parameters` here: https://github.com/near-daos/sputnik-dao-contract/blob/main/sputnikdao2/src/policy.rs#L270

I understand it as only the parameters you specify are updated, and the others are not touched. So I guess we don't have to send all parameters..

Regarding permissions. The test pass for adding the proposal, and there's also the [sandboxrpcplayground.js](https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/blob/e3c87a1cbad44313bca7fb9165022d6983c91896/sandboxrpc/sandboxrpcplayground.js) where I just played with the sputnik-dao contract to see how it works. As you can see there I add the proposal for changing the duration, and also vote for it, and you can see it takes effect. It seems that it works fine with the permission set up in that policy there. Would you take a look at that playground file and see if it makes sense?